core/src/main/scala/kafka/server/DynamicConfig.scala

```diff
@@ -30,10 +32,20 @@ import scala.jdk.CollectionConverters._
   * and can only be set dynamically.
   */
 object DynamicConfig {
+    object Broker {
+      private val brokerConfigs = {
+        val configs = QuotaConfigs.brokerQuotaConfigs()
+
+        // Filter and define all dynamic configurations
+        KafkaConfig.configKeys
+          .filter { case (configName, _) => AllDynamicConfigs.contains(configName) }
+          .foreach { case (_, config) => configs.define(config) }
+        configs
+      }
 
-  object Broker {
-    private val brokerConfigs = QuotaConfigs.brokerQuotaConfigs()
-    DynamicBrokerConfig.addDynamicConfigs(brokerConfigs)
+    // Non-dynamic properties are determined by subtracting dynamic broker config names from all config names.
```

Comment:
The key point is `DynamicBrokerConfig` should NOT have any `val` variables which are initialized by  `DynamicConfig.Broker`. Hence, the comments can be tweaked to 
```
In order to avoid circular reference, all DynamicBrokerConfig's variables which are initialized by `DynamicConfig.Broker` should be moved to `DynamicConfig.Broker`. Otherwise, those variables of DynamicBrokerConfig will see intermediate state of `DynamicConfig.Broker`, because `brokerConfigs` is created by `DynamicBrokerConfig.AllDynamicConfigs`
```

Reply:
I see, I will modify it soon.